File: backend/n2o/n2o-api/src/main/java/net/n2oapp/framework/api/metadata/global/view/widget/toolbar/Confirm.java
package net.n2oapp.framework.api.metadata.global.view.widget.toolbar;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import net.n2oapp.framework.api.metadata.Compiled;


/**
 * Модель Предупреждения при выполнении операций
 */

@Getter
@Setter
public class Confirm implements Compiled {
    @JsonProperty
    private String text;
    @JsonProperty
    private String title;
    @JsonProperty
    private String modelLink;
    @JsonProperty
    private String condition;
    @JsonProperty
    private Boolean closeButton;
    @JsonProperty
    private Boolean reverseButtons;
    @JsonProperty
    private ConfirmType mode;
    @JsonProperty
    private Button ok;
    @JsonProperty
    private Button cancel;

    @Getter
    @Setter
    public static class Button implements Compiled {
        @JsonProperty
        private String label;
        @JsonProperty
        private String color;

        public Button() {
        }

        public Button(String label, String color) {
            this.label = label;
            this.color = color;
        }
    }
}

File: frontend/n2o-framework/src/components/overlays/Confirm.ts
export enum ConfirmMode {
    POPOVER = 'popover',
    MODAL = 'modal',
}

type Color = 'info' | 'danger' | 'warning' | 'success' | 'primary' | 'secondary' | 'link'

export interface ConfirmJSON {
    text: string
    title: string
    modelLink: string
    condition: string
    closeButton: boolean
    reverseButtons: boolean
    mode: ConfirmMode
    ok: {
        label: string
        color: Color
    },
    cancel: {
        label: string
        color: Color
    }
}

export interface ConfirmProps {
    text: string
    title: string
    closeButton: boolean
    reverseButtons: boolean
    ok: {
        label: string
        color: Color
    },
    cancel: {
        label: string
        color: Color
    }
    id: string
    operation: { id: string, type: string }
    className?: string
    size: 'lg' | 'sm'
}
